### PR TITLE
[refactor] record migrations

### PR DESCRIPTION
**Pull request by steveruizok**
This PR removes comments from our record types, makes initial version optional, and unifies the order of initial / current version. 

- Initial versions are zero by default
- If no current version is provided to `defineMigrations`, migrations should be undefined
- Fixes TypeScript quirks in versioning (e.g. only initial version)

This PR also:
- Makes migrations optional when empty
- Removes reference to empty migrations

### Change Type

- [x] `major` — Breaking Change

### Test Plan

- [x] Unit Tests
- [ ] Webdriver tests

### Release Notes

- [tlschema] Improve `defineMigrations`
- [editor] Simplify migration definitions

### CODE REVIEW

File: packages/tlschema/src/assets/TLBookmarkAsset.ts [hunk 1]
```diff
@@ -14,7 +14,6 @@ export type TLBookmarkAsset = TLBaseAsset<
 	}
 >
 
-// --- VALIDATION ---
```
Comment (steveruizok) [inline packages/tlschema/src/assets/TLBookmarkAsset.ts:17]: There are a lot of these comments that I'm pulling out. We can have guides outside of the code.

File: packages/tlschema/src/assets/TLImageAsset.ts [hunk 2]
```diff
@@ -30,21 +29,14 @@ export const imageAssetTypeValidator: T.Validator<TLImageAsset> = createAssetVal
 	})
 )
 
-// --- MIGRATIONS ---
-// STEP 1: Add a new version number here, give it a meaningful name.
-// It should be 1 higher than the current version
 const Versions = {
-	Initial: 0,
```
Comment (steveruizok) [inline packages/tlschema/src/assets/TLImageAsset.ts:37]: in `defineMigrations`, Initial is now zero by default.

File: packages/tlstore/src/lib/test/defineMigrations.test.ts
Comment (steveruizok) [inline packages/tlstore/src/lib/test/defineMigrations.test.ts:1]: Adding a lot more tests here for the different combinations of props

File: packages/tlschema/src/createTLSchema.ts [hunk 3]
```diff
@@ -79,7 +85,13 @@ export function createTLSchema({
...
 		migrators: rootShapeTypeMigrations.migrators,
 		subTypeKey: 'type',
-		subTypeMigrations: Object.fromEntries(allShapeDefs.map((def) => [def.type, def.migrations])),
+		subTypeMigrations: Object.fromEntries(
+			(
+				allShapeDefs.filter((def) => def.migrations) as (CustomShapeTypeInfo & {
+					migrations: Migrations
+				})[]
+			).map((def) => [def.type, def.migrations])
+		),
```
Comment (ds300) [inline packages/tlschema/src/createTLSchema.ts:94]: I don't think it's safe to filter them out here, the migration code relies on the migrations metadata existing for all sub types. You could change this to return an 'empty' migrations object for shapes with no migrations defined.

**Pull request merged**